lib/ui/window/platform_configuration.cc

```diff
@@ -77,12 +76,31 @@ void PlatformConfiguration::DidCreateIsolate() {
   report_timings_.Set(tonic::DartState::Current(),
                       Dart_GetField(library, tonic::ToDart("_reportTimings")));
 
-  // TODO(loicsharma): This should only be created if the embedder enables the
-  // implicit view.
-  // See: https://github.com/flutter/flutter/issues/120306
-  windows_.emplace(kImplicitViewId,
-                   std::make_unique<Window>(
-                       kImplicitViewId, ViewportMetrics{1.0, 0.0, 0.0, -1, 0}));
+  library_.Set(tonic::DartState::Current(), library);
+}
+
+void PlatformConfiguration::AddView(int64_t view_id,
+                                    const ViewportMetrics& view_metrics) {
+  auto [window_iterator, insertion_happened] = windows_.emplace(
+      view_id, std::make_unique<Window>(library_, view_id, view_metrics));
+  FML_DCHECK(insertion_happened);
+  // Make the new window send an AddView message to Dart.
+  window_iterator->second->AddView();
```

Comment:
Currently this calls `PlatformDispatcher._addView` for all views, including the implicit view. The `PlatformDispatcher._addView` handles the implicit view specially by sending a window metric update as the implicit view was already created.

This seems to have two drawbacks:
1. It would be unexpected if `PlatformDispatcher._addView` was called twice for the implicit view. Currently, it is difficult to assert this
2. This sends an additional window metric update at startup. This might cause the framework to do unnecessary rebuilds.

An alternative approach might be to make `PlatformConfiguration::AddView` not call `PlatformDispatcher._addView` for the implicit view by not calling `Window::AddView`. We could then assert that `PlatformDispatcher._addView` is never called on the implicit view. What are your thoughts?

Reply:
> This sends an additional window metric update at startup

At startup, this function is called by `FlushRuntimeStateToIsolate` when the isolate is ready/ The accumulated states with one call for each view, either `AddView` or `UpdateWindowMetrics`.

Reply:
Marking this as resolved now that `PlatformDispatcher`'s constructor does not create the implicit view